### PR TITLE
feat(TextInputBox): Adding options and additional placeholder.

### PR DESCRIPTION
**Pull request by jakejung89**

### CODE REVIEW

File: src/components/TextInputBox/TextInputBox.js [hunk 1]
```diff
@@ -82,7 +110,20 @@ const InputBoxItem = styled(InputItem)`
...
-        const { label, name, inputType, error, disabled, collapsed, className, labelColor, lineColor } = this.props;
+        const {
+            label,
+            name,
+            inputType,
+            error,
+            disabled,
+            collapsed,
+            className,
+            promotedOptions,
```
Comment (cozzbp) [inline src/components/TextInputBox/TextInputBox.js]: Some of these new props should be added to default props and propTypes

Comment (jakejung89) [reply src/components/TextInputBox/TextInputBox.js]: Only new prop is placeholder. I'll go ahead and fix it.

**Comment by haronius** [comment]
There's a little bit of an overlap, might want to move the input out of the way a little
image

File: src/components/TextInputBox/TextInputBox.js [hunk 2]
```diff
@@ -121,7 +163,19 @@ export default class TextInputBox extends TextInput {
                     </TextBoxLabel>
                 }
                 </TextBox>
+                {options && <Caret onClick={this.toggleOptionsList} open={this.state.optionsListVisible} className={'pb-caret'} />}
```
Comment (haronius) [inline src/components/TextInputBox/TextInputBox.js]: What's the className do?

File: src/components/TextInputBox/TextInputBox.js [hunk 3]
```diff
@@ -121,7 +163,19 @@ export default class TextInputBox extends TextInput {
...
                 {this.renderHelperText()}
+                {options && <SelectOptions
+                    onOptionUpdate={this.onDropDownSelect}
+                    promotedOptions={promotedOptions || this.getPromotedOptions() }
+                    options={options}
+                    optionsCount={options.length}
+                    visible={this.state.optionsListVisible}
+                    width={this.props.selectOptionsWidth}
+                    optionsRef={(ref) => {
```
Comment (haronius) [inline src/components/TextInputBox/TextInputBox.js]: Could you add the scroll to top feature based in this ref

File: src/components/TextInputBox/TextInputBox.js (hunk 1 above)
Comment (jakejung89) [reply src/components/TextInputBox/TextInputBox.js]: done

File: src/components/TextInputBox/TextInputBox.js (hunk 2 above)
Comment (jakejung89) [reply src/components/TextInputBox/TextInputBox.js]: I think 'pb-caret' is used to run some tests.

**Pull request merged**